test/tokens/IpToken.t.sol

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.16;
+
+import "forge-std/Test.sol";
+import "../TestCommons.sol";
+import "../../contracts/tokens/IpToken.sol";
+import "../utils/TestConstants.sol";
+
+contract IpTokenTest is Test, TestCommons {
+    address internal _admin;
+    address internal _user1;
+    address internal _user2;
+    address internal _joseph;
+    address internal DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+
+    event Mint(address indexed account, uint256 amount);
+
+    function setUp() public {
+        _admin = vm.rememberKey(1);
+        _user1 = vm.rememberKey(2);
+        _user2 = vm.rememberKey(3);
+        _joseph = vm.rememberKey(4);
+    }
+
+    function testShouldTransferOwnershipSimpleCase1() public {
+        // given
+        IpToken ipToken = prepareIpToken();
+
+        // when
+        vm.prank(_admin);
+        ipToken.transferOwnership(_user1);
+        vm.prank(_user1);
+        ipToken.confirmTransferOwnership();
+
+        // then
+        assertEq(ipToken.owner(), _user1);
+    }
+
+    function testShouldNotTransferOwnershipSenderNotCurrentOwner() public {
+        // given - _admin as an initial owner
+        IpToken ipToken = prepareIpToken();
+
+        // when & then
+        vm.prank(_user1);
+        vm.expectRevert(abi.encodePacked("Ownable: caller is not the owner"));
+        ipToken.transferOwnership(_user1);
+    }
+
+    function testShouldNotConfirmTransferOwnershipSenderNotAppointedOwner() public {
+        // given
+        IpToken ipToken = prepareIpToken();
+
+        // when
+        vm.prank(_admin);
+        ipToken.transferOwnership(_user1);
+
+        // then
+        vm.prank(_user2);
+        vm.expectRevert(abi.encodePacked(IporErrors.SENDER_NOT_APPOINTED_OWNER));
+        ipToken.confirmTransferOwnership();
+    }
+
+    function testShouldNotConfirmTransferOwnershipTwiceSenderNotAppointedOwner() public {
+        // given
+        IpToken ipToken = prepareIpToken();
+
+        // when
+        vm.prank(_admin);
+        ipToken.transferOwnership(_user1);
+        vm.prank(_user1);
+        ipToken.confirmTransferOwnership();
+
+        // then
+        vm.prank(_user1);
+        vm.expectRevert(abi.encodePacked(IporErrors.SENDER_NOT_APPOINTED_OWNER));
+        ipToken.confirmTransferOwnership();
+    }
+
+    function testShouldNotTransferOwnershipSenderAlreadyLostOwnership() public {
+        // given
+        IpToken ipToken = prepareIpToken();
+
+        // when
+        vm.prank(_admin);
+        ipToken.transferOwnership(_user1);
+        vm.prank(_user1);
+        ipToken.confirmTransferOwnership();
+
+        // then
+        vm.prank(_admin);
+        vm.expectRevert(abi.encodePacked("Ownable: caller is not the owner"));
+        ipToken.transferOwnership(_user2);
+    }
+
+    function testShouldHaveRightsToTransferOwnershipSenderStillHaveRights() public {
+        // given
+        IpToken ipToken = prepareIpToken();
+
```

Comment:
Missing line in given section: ipToken.transferOwnership(_user1);

Reply:
it looks like a mistake in the hardhat test but I can do it like before

Reply:
transferOwnership

Reply:
No, it in on purpose, Ownable.transferOwnership from OpenZeppelin immediatelly change owner, here we wanted to show that went you executed transferOwnership first time, then you have rights to execute it again